src/gl/lineFeature.js

```diff
@@ -275,8 +286,8 @@ var gl_lineFeature = function (arg) {
           '  vec4 color = strokeColorVar;',
           allowDebug ? '  bool debug = bool(mod(fixedFlags, 2.0));' : '',
           '  float opacity = 1.0;',
-          '  int nearMode = int(info.x);',
-          '  int farMode = int(info.y);',
+          '  int nearMode = int(floor(info.x + 0.5));',
```

Comment:
why extra 0.5?

Reply:
is it because info.x is too small?

Reply:
We really want to use a `round` function, but WebGL doesn't have one.  In the osmesa implementation, this is sometimes very slightly less than the integer we start with (2.999999 versus 3), so a very tiny addition would be sufficient, but a general round function is often `floor(value + 0.5)`, so I've used that.

Reply:
Although the vertex shader produces consistent values (`info.x` at the three vertices will have an identical integer value in the range of [0-7]), this somehow is not always an exact integer in the fragment shader.  I don't know what osmesa does that causes this (the various actual GPUs that execute this seem to always have the exact integer), but this works around the issue.

Reply:
got it.